### PR TITLE
7.x self transform files and form select option

### PR DESCRIPTION
**Pull request by nhart**

### CODE REVIEW

File: transforms/dc_url_injection.xsl [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<xsl:stylesheet version="1.0"
```
Comment (jordandukart) [inline transforms/dc_url_injection.xsl]: Think these XSLs should live in the METRO module and then client configuration can make them available.

File: includes/handler.inc [hunk 2]
```diff
@@ -84,6 +84,26 @@ function islandora_oai_object_response_xml($params) {
...
 
   if (!empty($content)) {
+    $self_transform_file = variable_get("islandora_oai_self_transform_file_{$params['metadata_prefix']}", 'default');
+    if ($self_transform_file != 'default') {
+      $self_xslt = new XSLTProcessor();
+      $self_xsl = new DOMDocument();
+      $self_xsl->load($self_transform_file);
+      $self_input = new DOMDocument();
+
+      $url = url("islandora/object/{$object->id}", array(
```
Comment (jordandukart) [inline includes/handler.inc]: As opposed to passing the URL here explicitly I would suggest making a new hook to define an array of params available for XSLTs. That is METRO's module could then implement passing the URL parameter. In code we could do something like: 

`foreach ($params as $param) {
  $self_xslt->setParameter($param['namespace'], $param['name'], $param['value']);
}` or whatever structure you'd want to build

File: transforms/dc_url_injection.xsl (hunk 1 above)
Comment (nhart) [inline transforms/dc_url_injection.xsl]: Hum.. then why do we include any xslts in the islandora modules (eg: https://github.com/Islandora/islandora_xml_forms/tree/7.x/builder/self_transforms) ? a user can still make a copy of them and make whatever changes they want and hook them with their custom module.

Comment (jordandukart) [inline transforms/dc_url_injection.xsl]: Sure, but these transformations are not really standard at all and custom for the client.

**Pull request merged**